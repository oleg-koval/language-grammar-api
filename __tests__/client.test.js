const LanguageToolClient = require('../lib/index');

const mockedLogger = () => {
  return;
};

describe('LanguageToolClient:', () => {
  it('should have required functionality', () => {
    const dummyClient = new LanguageToolClient({ endpoint: 'https://example.com', log: mockedLogger });
    expect(dummyClient).toMatchSnapshot();
  });

  it('should be equal to function', () => {
    const dummyClient = new LanguageToolClient({ endpoint: 'https://example.com' });
    expect(dummyClient.languages).toMatchSnapshot();
  });

  it('should be equal to function', () => {
    const dummyClient = new LanguageToolClient({ endpoint: 'https://example.com' });
    expect(dummyClient.check).toMatchSnapshot();
  });

  it.only('Should throw error with missing options', () => {
    expect(() => new LanguageToolClient()).toThrowErrorMatchingSnapshot();
  });
});
