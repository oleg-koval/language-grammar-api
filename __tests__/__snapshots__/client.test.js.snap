// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageToolClient: Should throw error with missing options 1`] = `"missing options"`;

exports[`LanguageToolClient: should be equal to function 1`] = `[Function]`;

exports[`LanguageToolClient: should be equal to function 2`] = `[Function]`;

exports[`LanguageToolClient: should have required functionality 1`] = `
LanguageToolClient {
  "check": [Function],
  "client": [Function],
  "endpoint": "https://example.com",
  "languages": [Function],
  "log": [Function],
  "raw": [Function],
  "request": [Function],
}
`;
